docs/src/doc/contribution/knowledge-base/memory-management.md

```diff
@@ -0,0 +1,32 @@
+The purpose of this page is to give some clarification about how Godot and the JVM manage their memory together. 
+
+## Regular Godot memory management
+### Object and Refcounted
+Godot got two ways to manage memory. Classes belonging to the `Object` category are manually freed. If not done, it will remain in memory until the program terminates and leak. To make memory management easier, another category called `Refcounted` existed. They keep an internal counter of the numbers of how many times they are references by other parts of the program. It's only when reaching 0 that it will automatically be freed from memory.
```

Comment:
```suggestion
Godot got two ways to manage memory. Classes belonging to the `Object` category are manually freed. If not done, it will remain in memory until the program terminates and leak. To make memory management easier, another category called `Refcounted` exists. They keep an internal counter of the numbers of how many times they are references by other parts of the program. It's only when reaching 0 that it will automatically be freed from memory.
```